### PR TITLE
Improve attachments protection.

### PR DESCRIPTION
**Pull request by AliaksandrVahura-ScienceSoft**

### CODE REVIEW

File: laterpay/application/Controller/Frontend/Post.php [hunk 1]
```diff
@@ -496,6 +496,103 @@ public function check_user_access( $has_access, $post_id = null ) {
...
+                'attachment'
+            );
+            return $url;
+        }
+
+        return $url;
+    }
+
+    /**
+     * Prevent attachments from prepend paid content
```
Comment (drdla) [inline laterpay/application/Controller/Frontend/Post.php]: I have no idea what that means

File: laterpay/application/Controller/Frontend/Post.php [hunk 2]
```diff
@@ -496,7 +496,105 @@ public function check_user_access( $has_access, $post_id = null ) {
...
+            );
+
+            return $url;
+        }
+
+        return $url;
+    }
+
+    /**
+     * Prevent attachments from prepend paid content.
```
Comment (drdla) [inline laterpay/application/Controller/Frontend/Post.php]: What does "Prevent attachments from prepend paid content." mean?

Comment (AliaksandrVahura-ScienceSoft) [inline laterpay/application/Controller/Frontend/Post.php]: By default wordpress render attachment right before content, so it not included in the content and not covered by laterpay overlay/teaser. User can access this attachment easily by clicking on it. We should prevent such behaviour if this attachment is paid and user has no access to it. Filter "prepend_attachment" do this.

**Comment by mturalenka** [comment]
@AliaksandrVahura-ScienceSoft  please resolve conflicts

**Pull request merged**